File: problem1/fixed/B01069659.py
def maxValidWindowSum(nums, k):
    sum = 0
    neg = 0
    max = 0
    for i in range(0, k):
        sum += nums[i]
        if nums[i]<0:
            neg += 1
            if neg<=1:
                max = sum
                for i in range(1, len(nums)-k):
                    sum = sum + nums[i+k-1] - nums[i-1]
                    if nums[i+k-1] < 0:
                        neg += 1
                        if nums[i-1] < 0:
                            neg -= 1
                            if neg<=1 and sum>max:
                                max = sum
                                return max

File: problem1/java/B01069659.java
public class Solution {
{
public static int maxValidWindowSum(int[] nums, int k) {
    int sum = 0;
    int neg = 0;
    int max = 0;
    for(int i = 0; i<k; i++){
        sum += nums[i];
        if(nums[i]<0){
            neg++;
        }
    }
    if(neg<=1){
        max = sum;
    }
    for(int i = 1; i<nums.length-k; i++){
        sum = sum + nums[i+k-1] - nums[i-1];
        if(nums[i+k-1] < 0){
            neg++;
        }
        if(nums[i-1] < 0){
            neg--;
        }
        if(neg<=1 && sum>max)
            max = sum;
        }
    }
    return max;
}
}